cpp/oneapi/dal/backend/primitives/objective_function/logloss_dpc.cpp

```diff
@@ -29,25 +32,25 @@ sycl::event compute_probabilities(sycl::queue& q,
                                   const event_vector& deps) {
     const std::int64_t n = data.get_dimension(0);
     const std::int64_t p = data.get_dimension(1);
+
     ONEDAL_ASSERT(data.has_data());
     ONEDAL_ASSERT(parameters.has_data());
     ONEDAL_ASSERT(probabilities.has_mutable_data());
-    ONEDAL_ASSERT(parameters.get_dimension(0) == p + 1);
+
+    //ONEDAL_ASSERT(false);
+    ONEDAL_ASSERT(parameters.get_dimension(0) == fit_intercept ? p + 1 : p);
     ONEDAL_ASSERT(probabilities.get_dimension(0) == n);
 
     auto fill_event = fill<Float>(q, probabilities, Float(1), {});
     using oneapi::dal::backend::operator+;
 
-    auto param_arr = ndarray<Float, 1>::wrap(parameters.get_data(), 1);
-    Float w0 = fit_intercept ? param_arr.slice(0, 1).to_host(q, deps).at(0) : 0; // Poor perfomance
+    //auto param_arr = ndarray<Float, 1>::wrap(parameters.get_data(), 1);
 
-    auto event = gemv(q,
-                      data,
-                      parameters.get_slice(1, parameters.get_dimension(0)),
-                      probabilities,
-                      Float(1),
-                      w0,
-                      { fill_event });
+    Float w0 =
+        fit_intercept ? parameters.get_slice(0, 1).to_host(q, deps).at(0) : 0; // Poor perfomance
+    ndview<Float, 1> param_suf = fit_intercept ? parameters.get_slice(1, p + 1) : parameters;
```

Comment:
You can probably just use `auto` here

Reply:
There is a slight difference in usage of ndarray and ndview (for example when using functions get_slice or slice). So I think that it is better for code readability to specify the type.